vpr/src/place/weighted_median_move_generator.cpp

```diff
@@ -0,0 +1,178 @@
+        ClusterNetId net_id = cluster_ctx.clb_nlist.pin_net(pin_id);
+        if (cluster_ctx.clb_nlist.net_is_ignored(net_id))
+            continue;
+        if (int(cluster_ctx.clb_nlist.net_pins(net_id).size()) > placer_opts.place_high_fanout_net)
+            continue;
+        //all net pins are weighted with their own crititcalities
+        get_bb_cost_for_net_excluding_block(net_id, &coords, b_from, pin_id, criticalities, skip_net);
+        if (skip_net)
+            continue;
+
+        place_move_ctx.X_coord.insert(place_move_ctx.X_coord.end(), ceil(coords.xmin.criticality * 10), coords.xmin.edge);
+        place_move_ctx.X_coord.insert(place_move_ctx.X_coord.end(), ceil(coords.xmax.criticality * 10), coords.xmax.edge);
+        place_move_ctx.Y_coord.insert(place_move_ctx.Y_coord.end(), ceil(coords.ymin.criticality * 10), coords.ymin.edge);
+        place_move_ctx.Y_coord.insert(place_move_ctx.Y_coord.end(), ceil(coords.ymax.criticality * 10), coords.ymax.edge);
+    }
+
+    if ((place_move_ctx.X_coord.size() == 0) || (place_move_ctx.Y_coord.size() == 0))
+        return e_create_move::ABORT;
+
+    //calculate the weighted median region
+    std::sort(place_move_ctx.X_coord.begin(), place_move_ctx.X_coord.end());
+    std::sort(place_move_ctx.Y_coord.begin(), place_move_ctx.Y_coord.end());
+
+    if (place_move_ctx.X_coord.size() == 1) {
+        limit_coords.xmin = place_move_ctx.X_coord[floor((place_move_ctx.X_coord.size() - 1) / 2)];
+        limit_coords.xmax = limit_coords.xmin;
+    } else {
+        limit_coords.xmin = place_move_ctx.X_coord[floor((place_move_ctx.X_coord.size() - 1) / 2)];
+        limit_coords.xmax = place_move_ctx.X_coord[floor((place_move_ctx.X_coord.size() - 1) / 2) + 1];
+    }
+
+    if (place_move_ctx.Y_coord.size() == 1) {
+        limit_coords.ymin = place_move_ctx.Y_coord[floor((place_move_ctx.Y_coord.size() - 1) / 2)];
+        limit_coords.ymax = limit_coords.ymin;
+    } else {
+        limit_coords.ymin = place_move_ctx.Y_coord[floor((place_move_ctx.Y_coord.size() - 1) / 2)];
+        limit_coords.ymax = place_move_ctx.Y_coord[floor((place_move_ctx.Y_coord.size() - 1) / 2) + 1];
+    }
+
+    t_range_limiters range_limiters;
+    range_limiters.original_rlim = rlim;
+    range_limiters.dm_rlim = placer_opts.place_dm_rlim;
+    range_limiters.first_rlim = place_move_ctx.first_rlim;
+
+    t_pl_loc w_median_point;
+    w_median_point.x = (limit_coords.xmin + limit_coords.xmax) / 2;
+    w_median_point.y = (limit_coords.ymin + limit_coords.ymax) / 2;
+    if (!find_to_loc_centroid(cluster_from_type, from, w_median_point, range_limiters, to)) {
+        return e_create_move::ABORT;
+    }
+    return ::create_move(blocks_affected, b_from, to);
+}
+
+/* This routine finds the bounding box of a net from scratch excluding   *
+ * a specific block. This is very useful in some directed moves.         *
+ * It updates coordinates of the bb                                      */
```

Comment:
This comment should be expanded so the inputs and outputs of this function are clearly defined.  
I'd also move skip_net and coords beside each other (at the start or end of the parameter list) so inputs are separated from outputs.

Comment outline:
This routine finds the bounding box or a net, and the "cost" of each side of the bounding box, which is defined as the criticality of the connection that led to the bounding box extending that far.  If more than one terminal leads to a bounding box edge, we pick the cost using the criticality of the first one. This is helpful in computing weighted median moves.
Outputs: 
coords: the bounding box and the edge "costs"
skip_net:  ??

Inputs:
net_id: net we are considering
moving_pin_id:  pin (which should be on this net) on a block that is being moved.

Reply:
Done. Thanks for that suggestion, it is much clearer now.